src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -473,6 +480,127 @@ def process_scores_(
         # TODO: should we give num_entities in the constructor instead of inferring it every time ranks are processed?
         self.num_entities = num_entities
 
+    def evaluate_ogb(
+        self,
+        model: Model,
+        mapped_triples: MappedTriples,
+        batch_size: Optional[int] = None,
+        **kwargs,
+    ) -> MetricResults:
+        """
+        Evaluate a model using OGB's evaluator.
+
+        :param model:
+            the model; will be set to evaluation mode.
+        :param mapped_triples:
+            the evaluation triples
+
+            .. note ::
+                the evaluation triples have to match with the stored explicit negatives
+
+        :param batch_size:
+            the batch size
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`pykeen.nn.Model.predict`
+
+        :return:
+            the evaluation results
+
+        :raises ImportError:
+            if ogb is not installed
+        :raises NotImplementedError:
+            if `batch_size` is None, i.e., automatic batch size selection is selected
+        """
+        try:
+            from ogb.linkproppred import Evaluator as _OGBEvaluator
+        except ImportError as error:
+            raise ImportError("OGB evaluation requires `ogb` to be installed.") from error
+
+        if batch_size is None:
+            raise NotImplementedError("Automatic batch size selection not available for OGB evaluation.")
+
+        class _OGBEvaluatorBridge(_OGBEvaluator):
+            """A wrapper around OGB's evaluator to support evaluation on non-OGB datasets."""
+
+            def __init__(self):
```

Comment:
why is this necessary? to override and throw away the _actual_ `__init__`?

Reply:
exactly 😅